### PR TITLE
Fix emoji false-positives in invisible character detection

### PR DESCRIPTION
**Pull request by fiorsaoirse**

### CODE REVIEW

File: src/components/TextField/useTextValidation.ts [hunk 1]
```diff
@@ -44,19 +44,28 @@ export const useTextValidation = <T>({ value, onChange, invisibleCharacterWarnin
         state.current.detectedCodePoints = new Set();
     }, []);
     const detectionRegex = React.useMemo(() => chars.invisibleZeroWidthCharacters.createRegex(), []);
+    const segmenter = React.useMemo(() => new Intl.Segmenter(undefined, { granularity: "grapheme" }), []);
+    const emojiRegex = React.useMemo(() => new RegExp("\\p{Extended_Pictographic}|\\u20E3", "u"), []);
```
Comment (andreas-schultz) [inline src/components/TextField/useTextValidation.ts:48]: Why is the `u20E3` codepoint in this regex? This does not seem to be matched by the detection regex.

File: src/components/TextField/tests/useTextValidation.test.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        });
+
+        it("detects zero-width non-joiner (U+200C)", () => {
+            const callback = runWithValue("hello\u200Cworld");
+            expect(callback).toHaveBeenCalledWith(new Set([0x200c]));
+        });
+    });
+
+    describe("emoji false-positive prevention", () => {
+        it("does not flag ✔️ (base char + variation selector U+FE0F)", () => {
```
Comment (andreas-schultz) [inline src/components/TextField/tests/useTextValidation.test.tsx:56]: Maybe we should remove variation selectors completely from the black list. Emojis are not the only use case.

File: src/components/TextField/tests/useTextValidation.test.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            const callback = runWithValue("✔️");
+            expect(callback).toHaveBeenCalledWith(new Set());
+        });
+
+        it("does not flag ZWJ sequence emoji 👨‍👩‍👧‍👦", () => {
```
Comment (andreas-schultz) [inline src/components/TextField/tests/useTextValidation.test.tsx:61]: Same with the ZWJ. There are also legitimate use cases for it besides emojis.

File: src/components/TextField/useTextValidation.ts (hunk 1 above)
Comment (fiorsaoirse) [reply src/components/TextField/useTextValidation.ts:48]: `u20E3` is part of keycap emoji sequences like 1️⃣ (1 + uFE0F + u20E3). Without it, the variation selector `uFE0F` inside that cluster would trigger a false positive — since 1 is not `Extended_Pictographic`, the reg exp alone won't recognize it as an emoji to skip

File: src/components/TextField/tests/useTextValidation.test.tsx (hunk 2 above)
Comment (fiorsaoirse) [reply src/components/TextField/tests/useTextValidation.test.tsx:56]: Hi @haschek, what is your opinion here?

Comment (haschek) [reply src/components/TextField/tests/useTextValidation.test.tsx:56]: @andreas-schultz @fiorsaoirse I think it is ok to allow variation selectors.

What about extending `InvisibleCharacterWarningProps` with a a few boolean properties to allow a more detailed configuration regarding what is tested and what is allowed, or leading to calling the warning callback, e.g. `allowVariationSelectors?: boolean`. Then we also can have default values (`true` in this case). This is a transparent process, can come with good assumptions for the defaults and is extensible if we need to handle other ranges different from how we do now.

Comment (andreas-schultz) [reply src/components/TextField/tests/useTextValidation.test.tsx:56]: I would rather make this as good as it can be and if we missed something and the user gets a false positive, there should be at least enough information presented to the user on what the detected character is usually used for etc. Then the user could make a decision on which characters to remove.

Comment (andreas-schultz) [reply src/components/TextField/tests/useTextValidation.test.tsx:56]: Looking at the other edge cases I think most of them should still be reported, but the user should receive more information why they might be in the text in order to decide if they should be removed.

Comment (andreas-schultz) [reply src/components/TextField/tests/useTextValidation.test.tsx:56]: I've created a follow-up ticket.

**Review by andreas-schultz — DISMISSED** [review]

**Review by andreas-schultz — APPROVED** [review]
